trainer.py

```diff
@@ -15,7 +14,6 @@ def train(model, dataset, loss_function, epochs, optimizer):
             labels, thetas, xs, *targets = batch
             y_hat = model(xs, *thetas)
             loss = loss_function(y_hat, labels, thetas, *targets)
-            model.zero_grad() # from what I've read, this SHOULDN'T delete the computation graph
```

Comment:
I think this line should stay, otherwise the model gradients won't get reset each round. And the backwards in score_loss will effect the model parameters, so we need a zero_grad here.